### PR TITLE
[BugFix] avoid using after free when logging

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes # https://github.com/StarRocks/starrocks/issues/19249

## Problem Summary(Required) ：

the last log of the function use some alread freed memory after revert the PR https://github.com/StarRocks/starrocks/pull/18250

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/service/internal_service.cpp [hunk 1]
```diff
@@ -135,38 +135,52 @@ void PInternalServiceImplBase<T>::_transmit_chunk(google::protobuf::RpcControlle
...
-        LOG(WARNING) << "transmit_data failed, message=" << st.get_error_msg()
-                     << ", fragment_instance_id=" << print_id(request->finst_id()) << ", node=" << request->node_id();
-    }
-    if (done != nullptr) {
-        // NOTE: only when done is not null, we can set response status
-        st.to_protobuf(response->mutable_status());
-        done->Run();
-    }
-    VLOG_ROW << "transmit data: " << (uint64_t)(request) << " fragment_instance_id=" << print_id(request->finst_id())
-             << " node=" << request->node_id() << " cost time = " << MonotonicNanos() - begin_ts;
```
Comment (fzhedu) [inline be/src/service/internal_service.cpp:169]: after done, the request's members may be invalid.

File: be/src/service/internal_service.cpp [hunk 2]
```diff
@@ -135,38 +135,52 @@ void PInternalServiceImplBase<T>::_transmit_chunk(google::protobuf::RpcControlle
                                                   const PTransmitChunkParams* request, PTransmitChunkResult* response,
                                                   google::protobuf::Closure* done) {
     auto begin_ts = MonotonicNanos();
-    VLOG_ROW << "transmit data: " << (uint64_t)(request) << " fragment_instance_id=" << print_id(request->finst_id())
-             << " node=" << request->node_id() << " begin";
+    auto msg = "transmit data: " + std::to_string((uint64_t)(request)) +
+               " fragment_instance_id=" + print_id(request->finst_id()) +
+               " node = " + std::to_string(request->node_id());
```
Comment (liuyehcf) [inline be/src/service/internal_service.cpp:140]: " node = " -> " node="

Comment (stdpain) [reply be/src/service/internal_service.cpp:140]: concat log message to VLOG to avoid construct string when we disable VLOG

**Comment by fzhedu** [comment]
run starrocks_be_unittest

File: be/src/service/internal_service.cpp (hunk 2 above)
Comment (fzhedu) [reply be/src/service/internal_service.cpp]: fixed

**Comment by fzhedu** [comment]
run starrocks_be_unittest

File: be/src/service/internal_service.cpp [hunk 3]
```diff
@@ -135,38 +135,59 @@ void PInternalServiceImplBase<T>::_transmit_chunk(google::protobuf::RpcControlle
...
+    std::string transmit_info = "";
+    auto gen_transmit_info = [&transmit_info, &request]() {
+        transmit_info = "transmit data: " + std::to_string((uint64_t)(request)) +
+                        " fragment_instance_id=" + print_id(request->finst_id()) +
+                        " node=" + std::to_string(request->node_id());
+    };
+    if (VLOG_ROW_IS_ON) {
+        gen_transmit_info();
+    }
+    VLOG_ROW << transmit_info << " begin";
```
Comment (stdpain) [inline be/src/service/internal_service.cpp:147]: maybe unused

Comment (fzhedu) [reply be/src/service/internal_service.cpp:147]: it is also ok?

**Review by liuyehcf — APPROVED** [review]

File: be/src/service/internal_service.cpp [hunk 4]
```diff
@@ -135,38 +135,59 @@ void PInternalServiceImplBase<T>::_transmit_chunk(google::protobuf::RpcControlle
...
-            io_buf.copy_to(chunk->mutable_data(), chunk->data_size(), offset);
-            offset += chunk->data_size();
+            if (UNLIKELY(io_buf.size() < chunk->data_size())) {
+                auto msg = fmt::format("iobuf's size {} < {}", io_buf.size(), chunk->data_size());
+                LOG(WARNING) << msg;
+                st = Status::InternalError(msg);
+                return;
+            }
+            // also with copying due to the discontinuous memory in chunk
+            auto size = io_buf.cutn(chunk->mutable_data(), chunk->data_size());
```
Comment (stdpain) [inline be/src/service/internal_service.cpp:180]: why replace with cutn

Comment (fzhedu) [reply be/src/service/internal_service.cpp:180]: remove the `offset`, it is more simple.

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by silverbullet233 — APPROVED** [review]

**Pull request merged**